lib/option_converter.js

```diff
@@ -137,12 +137,12 @@ var contentFormatToString = function(value) {
   else if (value.length === 2)
     value = value.readUInt16BE(0)
   else
-    throw new Error('Content-Format option is too big')
+    return null
```

Comment:
In the case of an illegal format format, `null` is now returned which is handled well by the rest of the implementation. However, using the raw value or returning a `4.15` are also still on the table as options.

Reply:
4.15 would be a clean error ... feels better then null ... but honestly you are deeper in the specs/code then me right now

Reply:
Yeah, you are right. I think then it might be best to keep the code in this file as is, add another try-catch-block to `agent.js` (which is the part of the library that is crashing at the moment) and treat the content-format as `text/plain` if an illegal value is encountered. The server on the other hand could return `4.15` instead of `5.00` when it catches the exception. 

I think another part of the solution would be to increase the number of content-formats that are handled by default... I will have another look into this over the next couple of days :)

Reply:
I had another look into the [RFC](https://datatracker.ietf.org/doc/html/rfc7252#section-3) and noticed that the content-format is an "elective" option that has to be "silently ignored". I therefore changed the behavior of the agent and the server to either use the numeric value or to simply drop the option if its format is an "illegal" one. This is now also reflected in the test cases.

Regarding the additional options I would open another PR soon(ish) so that we could maybe also include this PR in the upcoming release and make the library a bit more stable :)

Reply:
makes sense